pkg/relay/submit.go

```diff
@@ -190,6 +202,14 @@ func (rs *Relay) storeSubmission(ctx context.Context, m *structs.MetricGroup, su
 	}
 	m.AppendSince(tPutHeader, "submitBlock", "putHeader")
 
+	if rs.config.Distributed && rs.config.StreamSubmissions {
+		select {
+		case rs.s.PublishStoreBlock() <- complete.Payload:
```

Comment:
wouldn't  `rs.s.PublishStoreBlock(ctx, complete.Payload)` 
and then 
```go
func (s *Client) PublishStoreBlock(ctx context.Context(), block something.Payload) error {
        select {
		case  s.storeRequests <-  block:
		case <-ctx.Done():
			return ctx.Err()
		}
}
```
be more readable?